submissions/Siusarna/data-structures/data-structures/linkedList.js

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable no-underscore-dangle */
+
+const Node = require('./Element.js');
+
+function getSuccessorNode(prevNode, successor) {
+  let successorNode = prevNode;
+  if (!successor) {
+    return successorNode;
+  }
+  // successor value saved in prevNode or prevNode._next() so we check these node
+  if (prevNode.getValue() === successor) {
+    successorNode = prevNode;
+  } else if (prevNode.getNext()) {
+    successorNode = prevNode.getNext();
+  } else {
+    throw new Error('not found this successor');
+  }
+  return successorNode;
+}
+
+module.exports = class LinkedList {
+  constructor() {
+    this._head = null;
+    this._size = 0;
+  }
+
+  searchForPrevNode(inputValue) {
+    let iteratorNode = this._head;
+    while (
+      iteratorNode.getNext() &&
+      iteratorNode.getNext().getValue() !== inputValue
+    ) {
+      iteratorNode = iteratorNode.getNext();
+    }
+    return iteratorNode;
+  }
+
+  insert(inputValue, successor = null) {
+    if (this._head === null) {
+      this._head = new Node(inputValue);
+      this._size += 1;
+    } else {
+      const prevNode = this.searchForPrevNode(successor);
+      const successorNode = getSuccessorNode(prevNode, successor);
+      const currentNode = new Node(inputValue, successorNode.getNext());
+      successorNode.setNext(currentNode);
+      this._size += 1;
+    }
+  }
+
+  remove(inputValue) {
+    if (this._head.getValue() === inputValue) {
+      this._head = this._head.getNext();
+      this._size -= 1;
+    } else {
+      const prevNodeForInputValue = this.searchForPrevNode(inputValue);
+      if (prevNodeForInputValue.getNext()) {
```

Comment:
```suggestion
      if (!prevNodeForInputValue.getNext()) {
        throw new Error('not found this value')
      }
```

Unnecessary `else` statement